src/components/LmsConfigurations/DegreedIntegrationConfigForm.jsx

```diff
@@ -0,0 +1,292 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import isEmpty from 'lodash/isEmpty';
+import {
+  ValidationFormGroup, Input, StatefulButton, Icon,
+} from '@edx/paragon';
+import { snakeCaseFormData } from '../../utils';
+import LmsApiService from '../../data/services/LmsApiService';
+import StatusAlert from '../StatusAlert';
+import SUBMIT_STATES from '../../data/constants/formSubmissions';
+import { handleErrors, validateLmsConfigForm } from './common';
+
+export const REQUIRED_DEGREED_CONFIG_FIELDS = [
+  'degreedBaseUrl',
+  'degreedUserId',
+  'degreedUserPassword',
+  'secret',
+  'degreedCompanyId',
+  'key',
+];
+
+function DegreedIntegrationConfigForm({ enterpriseId, config }) {
+  const [invalidFields, setInvalidFields] = useState({});
+  const [submitState, setSubmitState] = useState(SUBMIT_STATES.default);
+  const [active, setActive] = useState(config?.active);
+  const [error, setError] = useState(null);
+  const [curConfig, setConfig] = useState(null);
+
+  /**
+   * Creates a new third party provider configuration, then updates this list with the response.
+   * Returns if there is an error.
+   * @param {FormData} formData
+   */
+  const createDegreedConfig = async formData => {
+    const transformedData = snakeCaseFormData(formData);
+    transformedData.append('enterprise_customer', enterpriseId);
+    try {
+      const response = await LmsApiService.postNewDegreedConfig(transformedData);
+      return setConfig(response.data);
+    } catch (responseError) {
+      return handleErrors(responseError);
+    }
+  };
+
+  const updateDegreedConfig = async (formData, configId) => {
```

Comment:
I had thought about making the createLmsConfig functions to be reused across LMS configurations, but with them all using different LmsService API calls, it seemed like a headache to add that logic.

That said, the current function is *not* reused outside of this specific form. With the functional structure I even thought about not having a separate function and just using the logic where needed since we only use it once, but it's still cleaner this way.

Reply:
I wonder if it would make sense to have a generic LmsServiceApi (hook? module?) that accepts the endpoint, so the function itself doesn't have to make that choice. 🤔

Reply:
(Not a requested change, just something to think about)

Reply:
Yeah I think there's definitely room for refactoring down the line. May be something we end up doing anyways as part of the test/formate updates.